web/pdf_find_controller.js

```diff
@@ -86,20 +86,55 @@ const SPECIAL_CHARS_REG_EXP =
 const NOT_DIACRITIC_FROM_END_REG_EXP = /([^\p{M}])\p{M}*$/u;
 const NOT_DIACRITIC_FROM_START_REG_EXP = /^\p{M}*([^\p{M}])/u;
 
-let normalizationRegex = null;
+// The range [AC00-D7AF] corresponds to the Hangul syllables.
+// The few other chars are some CJK Compatibility Ideographs.
+const SYLLABLES_REG_EXP = /[\uAC00-\uD7AF\uFA6C\uFACF-\uFAD1\uFAD5-\uFAD7]+/g;
+const SYLLABLES_LENGTHS = new Map(
+  [
+    [...Array(0xd7af - 0xac00 + 1).keys()].map(x => x + 0xac00),
+    0xfa6c,
+    [...Array(0xfad1 - 0xfacf + 1).keys()].map(x => x + 0xfacf),
+    [...Array(0xfad7 - 0xfad5 + 1).keys()].map(x => x + 0xfad5),
+  ]
+    .flat()
+    .map(x => {
```

Comment:
Can't you use `flatMap` instead, since that ought to be a tiny bit more efficient?

Reply:
`flatMap` is in fact `mapFlat`....
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap